### PR TITLE
Closes #8233 Restrict API key usage by source IP

### LINKED ISSUE #8233: Restrict API key usage by source IP
### NetBox version

v3.1.4

### Feature type

Change to existing functionality

### Proposed functionality

Limit the access to the API with a certain API key by source IPs

### Use case

We have API keys that belong to 'service accounts'. The source IP of these services consumers are known.
We should be able to limit the usage of these service account API keys to only the known sources.
This cannot be done by firewalls but needs to go to the application layer

If an API key gets compromised, it cannot be used outside of that scope of source IPs

### Database changes

API Keys
- add multivalue 'Source IP'

### External dependencies

_No response_

### PR DESCRIPTION
**Pull request by PieterL75**

### CODE REVIEW

**Review by jeremystretch — CHANGES_REQUESTED** [review]
Looks like a great start! Thanks for taking this on.

File: docs/configuration/optional-settings.md [hunk 1]
```diff
@@ -277,6 +277,15 @@ Note that a plugin must be listed in `PLUGINS` for its configuration to take eff
 
 ---
 
+## PROXY_HEADER_REALIP
```
[the remark this answers is not in the document]
Comment (PieterL75) [reply docs/configuration/optional-settings.md:280]: I'll put the lines in remark, and create a separate FR for the new config variable name

**Review by PieterL75 — COMMENTED** [review]
Updating PR with changes

**Comment by PieterL75** [comment]
Noticed that I'm merging in the master, rather than the develop branch,,, 
Canceling this PR

**Pull request closed without merging**